apps/crn-server/test/data-providers/contentful/teams.data-provider.test.ts

```diff
@@ -0,0 +1,589 @@
+import { getContentfulGraphqlClientMock } from '../../mocks/contentful-graphql-client.mock';
+import { getContentfulEnvironmentMock } from '../../mocks/contentful-rest-client.mock';
+import {
+  TeamContentfulDataProvider,
+  teamUnreadyResponse,
+} from '../../../src/data-providers/contentful/teams.data-provider';
+import { getEntry } from '../../fixtures/contentful.fixtures';
+
+describe('Teams data provider', () => {
+  const contentfulGraphqlClientMock = getContentfulGraphqlClientMock();
+  const environmentMock = getContentfulEnvironmentMock();
+  const contentfulRestClientMock: () => Promise<Environment> = () =>
+    Promise.resolve(environmentMock);
+
+  const teamDataProvider = new TeamContentfulDataProvider(
+    contentfulGraphqlClientMock,
+    contentfulRestClientMock,
+  );
+
+  const contentfulGraphqlClientMockServer =
+    getContentfulGraphqlClientMockServer({
+      Teams: () => getContenfulGraphqlTeam(),
+    });
+
+  const teamDataProviderMock = new TeamContentfulDataProvider(
+    contentfulGraphqlClientMockServer,
+    contentfulRestClientMock,
+  );
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const getContentfulTeamDataObject = () => ({
+    ...getTeamDataObject(),
+    ...teamUnreadyResponse,
+  });
+
+  describe('Fetch method', () => {
+    test('Should fetch the list of teams from Contentful GraphQl', async () => {
+      const result = await teamDataProviderMock.fetch({});
+
+      expect(result.items[0]).toMatchObject(getContentfulTeamDataObject());
+    });
+
+    test('Should return an empty result when no teams exist', async () => {
+      const contentfulGraphQLResponse = getContentfulTeamsGraphqlResponse();
+      contentfulGraphQLResponse.teamsCollection!.total = 0;
+      contentfulGraphQLResponse.teamsCollection!.items = [];
+
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+        contentfulGraphQLResponse,
+      );
+
+      const result = await teamDataProvider.fetch({});
+
+      expect(result).toEqual({
+        items: [],
+        total: 0,
+      });
+    });
+
+    test('Should throw an error with a specific error message when the graphql client throws one', async () => {
+      contentfulGraphqlClientMock.request.mockRejectedValueOnce(
+        new GraphQLError('some error message'),
+      );
+
+      await expect(teamDataProvider.fetch({})).rejects.toThrow(
+        'some error message',
+      );
+    });
+
+    test('Should return an empty result when the query is returned as null', async () => {
+      const contentfulGraphQLResponse = getContentfulTeamsGraphqlResponse();
+      contentfulGraphQLResponse.teamsCollection = null;
+
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce(
+        contentfulGraphQLResponse,
+      );
+
+      const result = await teamDataProvider.fetch({});
+
+      expect(result).toEqual({
+        items: [],
+        total: 0,
+      });
+    });
+
+    test('Should return teams', async () => {
```

Comment:
How is this test different to the first test `Should fetch the list of teams from Contentful GraphQl`?

Reply:
The `teamDataProvider` and `teamDataProviderMock` are using different mocks as graphql client one of them uses https://github.com/yldio/asap-hub/blob/CRN-1250-create-teams-content-model-in-contentful/packages/contentful/src/mocks/graphql-client.mock.ts#L7 and the other https://github.com/yldio/asap-hub/blob/CRN-1250-create-teams-content-model-in-contentful/apps/crn-server/test/mocks/contentful-graphql-client.mock.ts#L3

Reply:
to be more clear, one of them is testing the graphql query and the other is not 😉

Reply:
It's testing that given a valid Contentful response provided by `contentfulGraphqlClientMock`, the data provider returns the expected response. But ok, I admit it's redundant. I wouldn't mind having them both. But I can remove this one if you prefer.